File: count-the-digits-that-divide-a-number-2520/Solution.java

/**
 * LeetCode problem # 2520, 'Count the Digits That Divide a Number'
 * https://leetcode.com/problems/count-the-digits-that-divide-a-number/
 */

class Solution {
    public int countDigits(int num) {
        int res = 0;
        int numCopy = num;
        while (num != 0) {
            int digit = num % 10;
            num /= 10;
            if (digit != 0 && numCopy % digit == 0) {
                ++res;
            }
        }
        return res;
    }
}

File: count-the-digits-that-divide-a-number-2520/solution.js
/**
 * LeetCode problem # 2520, 'Count the Digits That Divide a Number'
 * https://leetcode.com/problems/count-the-digits-that-divide-a-number/
 */
/**
 * @param {number} num
 * @return {number}
 */
function countDigits(num) {
  let res = 0;
  const digits = getDigits(num);
  for (const digit of digits) {
    if (num % digit === 0) {
      ++res;
    }
  }
  return res;
}

function getDigits(num) {
  return (num + "").split("").map((d) => parseInt(d));
}
